app/controllers/api/service_templates_controller.rb

```diff
@@ -20,6 +20,10 @@ def create_resource(_type, _id, data)
 
     def edit_resource(type, id, data)
       catalog_item = resource_search(id, type, collection_class(:service_templates))
+      if data.key?("schedule_time")
+        schedule_time = data&.delete("schedule_time")
+        catalog_item.update_schedule(schedule_time)
```

Comment:
It feels like this code should be responsible for calling the "update or delete schedules" method if the `schedule_time` is provided or not.  See comment [here](https://github.com/ManageIQ/manageiq/pull/17645#discussion_r199217984)